diff --git a/src/libbluray/bdj/bdj.c b/src/libbluray/bdj/bdj.c
index 1cb1bfe..af3617d 100644
--- a/src/libbluray/bdj/bdj.c
+++ b/src/libbluray/bdj/bdj.c
@@ -869,6 +869,10 @@
         "com.aacsla.bluray.online",
         "com.aacsla.bluray.mc",
         "com.aacsla.bluray.mt",
+        "net.kodinerds.maven.kodi",
+        "net.kodinerds.maven.kodi20",
+        "net.kodinerds.maven.kodi.firetv",
+        "net.kodinerds.maven.kodi20.firetv",
         "org.videolan.backdoor", /* entry for injected Xlet / runtime fixes */
 };
 static const size_t num_java_base_exports = sizeof(java_base_exports) / sizeof(java_base_exports[0]);
@@ -903,9 +907,13 @@
     option[n++].optionString = str_dup   ("-Dawt.toolkit=java.awt.BDToolkit");
     option[n++].optionString = str_dup   ("-Djava.awt.graphicsenv=java.awt.BDGraphicsEnvironment");
     option[n++].optionString = str_dup   ("-Djava.awt.headless=false");
-    option[n++].optionString = str_dup   ("-Xms256M");
-    option[n++].optionString = str_dup   ("-Xmx256M");
-    option[n++].optionString = str_dup   ("-Xss2048k");
+    option[n++].optionString = str_dup   ("-Xms512M");
+    option[n++].optionString = str_dup   ("-Xmx512M");
+    option[n++].optionString = str_dup   ("-Xss4096k");
+    option[n++].optionString = str_printf("-Djava.io.tmpdir=%s", java_home);
+    BD_DEBUG(DBG_BDJ | DBG_CRIT, "Use java_home as cache dir.\n");
+    option[n++].optionString = str_dup   ("-Djava.awt.headless=false");
+    BD_DEBUG(DBG_CRIT | DBG_BDJ, "Disable X11 check\n");
 #ifdef HAVE_BDJ_J2ME
     option[n++].optionString = str_printf("-Djava.home=%s", java_home);
     option[n++].optionString = str_printf("-Xbootclasspath/a:%s/lib/xmlparser.jar", java_home);
